client/my-sites/add-ons/hooks/use-add-ons.ts

```diff
@@ -140,7 +140,7 @@ const useAddOns = ( siteId?: number ): ( AddOnMeta | null )[] => {
 
 		const spaceUpgradesPurchased: number[] = [];
 
-		if ( filteredTransactions?.length ) {
+		if ( filteredTransactions?.length && siteId ) {
```

Comment:
I'm not sure I understand why the siteId all of a sudden is needed here. Could you help me understand?

Reply:
Thanks for the patience! This was meant to fix a bug when we call `use-add-ons` in the context of onboarding. For some reason, add-on data was incorrectly deciding that both 50GB add-ons and 100GB add-ons were _already_ purchased, even though no site had been created yet.

<img width="1487" alt="Screenshot 2023-09-15 at 1 29 29 AM" src="https://github.com/Automattic/wp-calypso/assets/5414230/48651b84-9520-4504-a9fa-9de16ad19adc">

To elaborate on why, `filteredTransactions` returns transactions for a given site when `siteId` exists. When, however, siteId is undefined, `filteredTransactions` fetches _all_ transactions for a user, regardless of what site the purchase was made on. 

Since I had made _some_ space upgrade add-on purchase in the past on _any_ site, the logic then proceeded to mark add-on data as `purchased`

https://github.com/Automattic/wp-calypso/blob/426d78e95c58f31f2d1f9c19ade66e4dd33a6504/client/my-sites/add-ons/hooks/use-add-ons.ts#L231-L244

Reply:
Thanks for the explanation. It looks like the transaction data is used for filtering out the storage add-ons that have lower volumn than the owned one. Also, how the whole transaction data is fetched and filtered is actually quite non-ideal –– it fetches the whole transaction history and than filter from the client side, while all these should be done from the server side so our users don't have to waste their bandwidth and local computational resources to perform all these. Since correcting the whole thing would be out of scope, let's start by not fetching those transaction data in the onboarding flow since that doesn't make sense. It'd involve some level of decoupling, but we will have a cleaner code without tricky conditions needing explicit documentation :)

Reply:
I took a very simplistic approach to this https://github.com/Automattic/wp-calypso/pull/81340/commits/0d13a0e40a572166f2e33fa2796bfed140648dda